pkg/controller/import-controller.go

```diff
@@ -157,6 +191,49 @@ func (ic *ImportController) processPvcItem(pvc *v1.PersistentVolumeClaim) error
 	return nil
 }
 
+func (ic *ImportController) createImporterPod(pvc *v1.PersistentVolumeClaim, pvcKey string) error {
+	var scratchPvc *v1.PersistentVolumeClaim
+	var scratchPvcName *string
+	var err error
+
+	requiresScratch := ic.requiresScratchSpace(pvc)
+	if requiresScratch {
+		name := pvc.Name + "-scratch"
+		scratchPvcName = &name
+	}
+
+	podEnvVar, err := createImportEnvVar(ic.clientset, pvc)
+	if err != nil {
+		return err
+	}
+
+	// all checks passed, let's create the importer pod!
+	ic.expectPodCreate(pvcKey)
+	pod, err := CreateImporterPod(ic.clientset, ic.image, ic.verbose, ic.pullPolicy, podEnvVar, pvc, scratchPvcName)
+	if err != nil {
+		ic.observePodCreate(pvcKey)
+		return err
+	}
+	if requiresScratch {
+		scratchPvc, err = ic.findScratchPvcFromCache(pvc)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return err
+		}
+		if scratchPvc == nil {
+			storageClassName, err := GetScratchPvcStorageClass(ic.clientset, pvc)
+			if err != nil {
+				return err
+			}
+			// Scratch PVC doesn't exist yet, create it. Determine which storage class to use.
+			scratchPvc, err = CreateScratchPersistentVolumeClaim(ic.clientset, pvc, pod, storageClassName)
+			if err != nil {
```

Comment:
I believe that if this fails, the pod will be pending forever.  Somewhere you you have to make sure that the scratch pvc exists so long as it is needed and the pod exists.

I initially thought you could simply delete the pod if creating the pvc fails.  That may be nice but is not sufficient.  The controller could die between operations.

Reply:
How about in the main loop, I check if pod exists and is in pending, and scratch PVC doesn't exist, I try to create it again there